### PR TITLE
UHF-5341: Remove TPR Units not located in Helsinki

### PR DESCRIPTION
**Pull request by ltpk**
How to test:

- set up any instance
- get this branch of the module: `composer require drupal/helfi_platform_config:dev-UHF-5341_remove_non-helsinki_tpr_units`
- import all TPR Units: `drush mim tpr_unit`
- see a large number of TPR Units in the admin view
- run the drush command `drush helfi:remove-non-helsinki-tpr-units`
- see a considerably smaller number of TPR Units in the admin view

### CODE REVIEW

**Review by jeremysteerio — APPROVED** [review]
Works as described (unit count dropped to 28k from 46k).

File: src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+   */
+  protected EntityTypeManagerInterface $entityTypeManager;
+
+  /**
+   * Constructs a new instance.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
```
Comment (tuutti) [inline src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php:28]: Use constructor property promotion: https://stitcher.io/blog/constructor-promotion-in-php-8

File: src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        $lock_service->release($unit->id(), $unit->language()->getId(), NULL, NULL, 'tpr_unit');
+      }
+
+      // Force delete unit.
+      $unit->delete(TRUE);
+
+      $unit_count++;
+    }
+
+    \Drupal::messenger()->addMessage($unit_count . ' units deleted.');
```
Comment (tuutti) [inline src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php:69]: Use `$this->output()->writeln()`.

File: src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    $units = $entityStorage->loadMultiple($unit_ids);
+
+    // Set up content lock service.
+    $lock_service = \Drupal::service('content_lock');
+
+    $unit_count = 0;
+
+    // Delete the units.
+    foreach ($units as $unit) {
+      \Drupal::messenger()->addMessage('Deleting "' . $unit->label() . '"');
```
Comment (tuutti) [inline src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php:56]: Use `$this->output()->writeln()`.

File: src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+      // Release content lock if needed.
+      if ($lock_service->fetchLock($unit->id(), $unit->language()->getId(), NULL, 'tpr_unit')) {
```
Comment (tuutti) [inline src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php:59]: Use named arguments here to skip optional/default paramers (https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments):
`$lock_service->fetchLock($unit->id(), $unit->language()->getId(), entity_type: 'tpr_unit')`

File: src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        $lock_service->release($unit->id(), $unit->language()->getId(), NULL, NULL, 'tpr_unit');
```
Comment (tuutti) [inline src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php:60]: Use named arguments here to skip optional/default paramers (https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments):
`$lock_service->release($unit->id(), $unit->language()->getId(), entity_type: 'tpr_unit')`

**Review by tuutti — CHANGES_REQUESTED** [review]

File: src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    $unit_ids = $entityStorage->getQuery()
+      ->condition('address__locality', 'Helsinki', '!=')
+      ->condition('address__locality', 'Helsingfors', '!=')
+      ->execute();
+
+    $units = $entityStorage->loadMultiple($unit_ids);
+
+    // Set up content lock service.
+    $lock_service = \Drupal::service('content_lock');
```
Comment (tuutti) [inline src/Commands/RemoveNonHelsinkiTPRUnitsCommands.php:50]: Inject `content_lock` service via drush.services.yml/constructor

**Review by tuutti — CHANGES_REQUESTED** [review]

**Review by tuutti — APPROVED** [review]

**Review by tuutti — CHANGES_REQUESTED** [review]
Nevermind. I think we need to add update hook that enables `content_lock` module since it's not done automatically.

**Review by tuutti — APPROVED** [review]

**Pull request merged**